File: MainApp/src/main/java/com/sleticalboy/dailywork/data/DataEngine.java
package com.sleticalboy.dailywork.data;

/**
 * Created on 20-3-27.
 *
 * @author binlee sleticalboy@gmail.com
 */
public final class DataEngine {

    private static DataEngine sDataEngine;
    private IndexModel mIndexModel;

    private DataEngine() {
        // no instance
    }

    public static DataEngine get() {
        if (sDataEngine == null) {
            sDataEngine = new DataEngine();
        }
        return sDataEngine;
    }

    public IndexModel indexModel() {
        if (mIndexModel == null) {
            mIndexModel = new IndexModel();
        }
        return mIndexModel;
    }
}

File: MainApp/src/main/java/com/sleticalboy/learning/data/DataEngine.kt
package com.sleticalboy.learning.data

/**
 * Created on 20-3-27.
 *
 * @author binlee sleticalboy@gmail.com
 */
class DataEngine private constructor() {

  private var mIndexModel: IndexModel? = null

  fun indexModel(): IndexModel {
    if (mIndexModel == null) {
      mIndexModel = IndexModel()
    }
    return mIndexModel!!
  }

  companion object {

    private val sDataEngine = DataEngine()

    fun get(): DataEngine = sDataEngine
  }
}